stock_picking_force_assign/models/stock_picking.py

```diff
@@ -10,18 +10,24 @@ class StockPicking(models.Model):
     def action_force_assign_pickings(self):
         """ Unreserve other pickings in order to reserve pickings in self """
         link_template = '<a href="#" data-oe-model="%s" data-oe-id="%d">%s</a>'
+        if self.state == 'waiting':
+            self._unchain_waiting(link_template)
         to_unreserve = self._force_assign_find_moves()
-        to_unreserve._do_unreserve()
-        self.message_post(body=_(
-            'Unreserved picking(s) %s in order to assign this one'
-        ) % ', '.join(to_unreserve.mapped('picking_id').mapped(
-            lambda x: link_template % (x._name, x.id, x.name)
-        )))
-        to_unreserve.mapped('picking_id').message_post(body=_(
-            'Unreserved this picking in order to assign %s'
-        ) % ', '.join(self.mapped(
-            lambda x: link_template % (x._name, x.id, x.name)
-        )))
+        if to_unreserve:
+            to_unreserve.mapped(lambda m: m._do_unreserve())
+            self.message_post(body=_(
+                'Unreserved picking(s) %s in order to assign this one'
+            ) % ', '.join(to_unreserve.mapped('picking_id').mapped(
+                lambda x: link_template % (x._name, x.id, x.name)
+            )))
+            to_unreserve.mapped('picking_id').mapped(lambda p: p.message_post(body=_(
+                'Unreserved this picking in order to assign %s'
+            ) % ', '.join(self.mapped(
+                lambda x: link_template % (x._name, x.id, x.name)
+            ))))
+            # Set procure_method of the affected moves to make_to_stock to not get stuck
+            if not all(method == 'make_to_stock' for method in to_unreserve.mapped('procure_method')):
+                to_unreserve.write({'procure_method': 'make_to_stock'})
```

Comment:
```suggestion
            to_unreserve.filtered(
                lambda x: x.procure_method == 'make_to_order'
            ).write({
                'procure_method': 'make_to_stock'
            })
```

this avoids unnecessary writes to the database, is agnostic against introducing make_to_something_else in the future and a nice side effect is that you don't need the conditional above, as all of this are no-ops when there's nothing to unreserve